src/Compiler/CodeGen/IlxGen.fs

```diff
@@ -10692,7 +10696,15 @@ and GenTypeDef cenv mgbuf lazyInitInfo eenv m (tycon: Tycon) =
                             let ilPropName = fspec.LogicalName
                             let ilMethName = "get_" + ilPropName
                             let access = ComputeMemberAccess isPropHidden
-                            yield mkLdfldMethodDef (ilMethName, access, isStatic, ilThisTy, ilFieldName, ilPropType)
+                            let isStruct = tcref.IsFSharpStructOrEnumTycon && not tcref.IsEnumTycon
+
+                            let attrs =
+                                if isStruct && not isStatic then
```

Comment:
I guess this is because the optimization won't optimize much for the static case?

Reply:
Yeah C# also doesn't add the attribute in that case (even if it is read only).